app/src/main/java/ca/bc/gov/bchealth/ui/healthpass/HealthPassFragment.kt

```diff
@@ -67,30 +66,22 @@ class HealthPassFragment : Fragment(R.layout.fragment_helath_pass) {
             when (it) {
                 BioMetricState.SUCCESS -> {
                     viewModel.onAuthenticationRequired(false)
-                    if (viewModel.isBcscLoginRequiredPostBiometrics) {
-                        sharedViewModel.destinationId = 0
-                        checkLogin()
-                        viewModel.isBcscLoginRequiredPostBiometrics = false
-                    }
+                    viewModel.launchCheck()
                 }
                 else -> {
                     findNavController().popBackStack()
                 }
             }
         }
 
-        val savedStateHandle: SavedStateHandle =
-            findNavController().currentBackStackEntry!!.savedStateHandle
-        savedStateHandle.getLiveData<Boolean>(
+        findNavController().currentBackStackEntry?.savedStateHandle?.getLiveData<Boolean>(
             BCSC_AUTH_SUCCESS
-        )
-            .observe(viewLifecycleOwner, {
-                if (it) {
-                    savedStateHandle.set(BCSC_AUTH_SUCCESS, false)
-                    if (sharedViewModel.destinationId > 0)
-                        findNavController().navigate(R.id.addCardOptionFragment)
-                }
-            })
+        )?.observe(viewLifecycleOwner, {
+            if (it) {
+                findNavController().currentBackStackEntry?.savedStateHandle
+                    ?.set(BCSC_AUTH_SUCCESS, false)
+            }
```

Comment:
Your forgot to add this condition back 
if (sharedViewModel.destinationId > 0)
                        findNavController().navigate(R.id.addCardOptionFragment)

Reply:
As discussed navigation is not required here as we stay on the same fragment in the on boarding flow